src/rpc/dial.rs

```diff
@@ -443,6 +537,54 @@ impl DialBuilder<WithCredentials> {
             .layer(AddAuthorizationLayer::bearer(&token))
             .service(channel))
     }
+
+    async fn create_channel(
+        allow_downgrade: bool,
+        domain: &str,
+        uri: Uri,
+        for_mdns: bool,
+    ) -> Result<Channel> {
+        let mut chan = Channel::builder(uri.clone());
+        if for_mdns {
+            let tls_config = ClientTlsConfig::new().domain_name(domain);
+            chan = chan.tls_config(tls_config)?;
+        }
+        let chan = match chan
+            .connect()
+            .await
+            .with_context(|| format!("Connecting to {:?}", uri.clone()))
+        {
+            Ok(c) => c,
+            Err(e) => {
+                if allow_downgrade {
+                    let mut uri_parts = uri.clone().into_parts();
+                    uri_parts.scheme = Some(Scheme::HTTP);
+                    let uri = Uri::from_parts(uri_parts)?;
+                    Channel::builder(uri.clone()).connect().await?
+                } else {
+                    return Err(anyhow::anyhow!(e));
+                }
+            }
+        };
+        Ok(chan)
+    }
+
+    /// attempts to establish a connection with credentials to the DialBuilder's given uri
+    pub async fn connect(self) -> Result<AddAuthorization<ViamChannel>> {
+        let original_uri = match self.duplicate_uri() {
+            Some(uri) => uri,
+            None => {
+                return Err(anyhow::anyhow!(
+                    "Attempting to connect but there was no uri"
+                ))
+            }
+        };
+        let mdns_uri = webrtc::action_with_timeout(self.get_mdns_uri(), Duration::from_secs(5))
```

Comment:
I think we should reflect the same dial options go has to possible disable this. also matching its logging style feels needed to know how we are connecting

Reply:
we have `disable_mdns` in the options. if a user sets it, then `get_mdns_uri()` immediately returns a `None` and we don't try to use it. is this what you mean?

will add log statements re: how we are connecting.

Reply:
ah okay! I'm a bad reader!

Reply:
added some log statements!